### ISSUE TITLE
Distribute binaries outside of play store

### ISSUE THREAD

**Issue by Derkades** [minvws/nl-covid19-coronacheck-app-android#24]
### Describe the bug, issue or concern

Please consider providing the application outside of the Google Play Store, for example by adding apk files to GitHub releases or ideally adding the app to [F-Droid](https://f-droid.org/). Some people can't or don't want to use Google Play Services on their Android devices, for example because of privacy concerns.

Contact tracing and/or checking apps for [France](https://f-droid.org/en/packages/fr.gouv.android.stopcovid) ([and third party](https://f-droid.org/en/packages/fr.covidat.android/)), [Switzerland](https://f-droid.org/en/packages/ch.corona.tracing/), [United Kingdom](https://f-droid.org/en/packages/covidsecure.uk.venuecheckin/) and [Germany](https://f-droid.org/en/packages/de.corona.tracing/) are on F-Droid as well.

### Governance
- [x] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

**Comment by pjvleeuwen** [comment]
I would like to add my voice to this ticket: I do not own a Google account with which to login to Google's Play Store, nor does my Android phone have the Google Services Framework. I am of the (rather strong) opinion that it is fundamentally wrong that a person is forced to provide any personal information to any oversea's tech giant to be able to participate in products build with tax money for which there could never be a competing alternative. I would like to ask you kindly to provide an impartial and secure source for citizens to download this app. Thank you for your consideration!

**Comment by obfusk** [comment]
The Swiss COVID Certificate Apps should be available in F-Droid soon as well: https://github.com/admin-ch/CovidCertificate-App-Android/issues/40; and because they use Reproducible Builds, the F-Droid version will be completely identical to the official one (including the signature), which also adds a bit of extra security (since it can be verified that the APK matches the published source code).

**Comment by obfusk** [comment]
https://fd.nl/economie-politiek/1383007/overheid-dwingt-burger-in-de-richting-van-google-en-apple

**Comment by diederikdehaas** [comment]
I'd like to add my voice/support for this ticket as well. The people most interested in the source code are often also FLOSS advocates and thus have a problem with Google('s proprietary services/software).

At https://coronacheck.nl/nl/faq there are several questions wrt a Dutch and an international proof (of vaccination).
I'm pretty sure the same/similar question applies to other countries in the EU.
So that begs the question: why is this app (and similar ones) not developed in collaboration/coordination with other (FLOSS) developers around the EU? Ideally resulting in a/one completely FLOSS app* usable throughout the EU.

*) In Debian terms that would mean, suitable for the `main` archive section, thus not only the app is Free Software, but also its dependencies.

**Comment by Derkades** [comment]
@pjvleeuwen 
> I am of the (rather strong) opinion that it is fundamentally wrong that a person is forced to provide any personal information to any oversea's tech giant to be able to participate in products build with tax money for which there could never be a competing alternative.

It's possible to generate a certificate in a browser. Having the app on F-Droid would be convenient but I wouldn't say we're "forced" to use the Play Store.

English: https://coronacheck.nl/en/print/ Dutch: https://coronacheck.nl/nl/print/ 

It's advertised as a paper certificate for people without smartphones, but keeping it digitally as a PDF should be fine as well.

**Comment by obfusk** [comment]
> It's possible to generate a certificate in a browser. Having the app on F-Droid would be convenient but I wouldn't say we're "forced" to use the Play Store.

That's only one side of the verification process. AFAIK if you want to verify someone's QR code (printed or not), you still need the app.

**Comment by dutchcodes** [comment]
+1 for this ticket. I'll write this in Dutch to not give out a false impression that it only is an issue for international/English speaking people but is also hindering adoption for Dutch speaking citizens.

Er zijn vele redenen waarom mensen de Google Play Store niet gebruiken. Bijvoorbeeld Huawei gebruikers die verminderde toegang hebben, mensen die hun oude mobiel een nieuw leven hebben gegeven door er een alternatief Android OS (bijv LineageOS) op te zetten waardoor beveiligingsupdates nog steeds binnen komen. Ook zijn er mensen die bewust compleet van het gebruik van Google afstappen en mobiele telefoons kopen gebaseerd op Android (Google) maar dan zonder Google Play Store of Services inbegrepen. Voorbeelden zijn de besturingssystemen /e/, CalyxOS, GrapheneOS etc.

Het is fijn dat deze app wél te installeren is en werkt zonder geinstalleerde Google Services-API in tegenstelling tot de Corona Melders app maar.... zoals ook in deze issue beschreven, zorg er voor dat _iedereen_ deze app kan installeren. Losse APK's of gebruik van F-Droid (voorkeur), zo bereik je iedereen als overheid.

**Comment by pjvleeuwen** [comment]
@Derkades 
> It's possible to generate a certificate in a browser. Having the app on F-Droid would be convenient but I wouldn't say we're "forced" to use the Play Store.

Thanks for providing an alternative. It just makes me sad that developers make assumptions like all users using google. The swimming pool that teaches my daughter swimming only communicates important notifications via Facebook, even though they have a stand-alone website. Then they cannot store my email address in their system because the TLD is more then 3 characters. This is just another example in a long list. Even the government! I would expect the government to have internal development standards that stive for maximum inclusiveness. I just keep hoping it is ignorence and not a concious decision to disregard a portion of the user base.

@dutchcodes 
PS, Nederlands is ook mijn moerstaal, maar als het over dev-aangelegenheden gaat switch ik automatisch naar Engels... :-)

**Comment by obfusk** [comment]
> https://fd.nl/economie-politiek/1383007/overheid-dwingt-burger-in-de-richting-van-google-en-apple

Een paar citaten:

> * Apps van de Rijksoverheid worden alleen aangeboden voor klanten van Google en Apple.
> * Daarmee belemmeren ze concurrentie en worden burgers gedwongen in zee te gaan met de Amerikaanse techbedrijven.
> * Volgens experts zijn er verschillende veilige alternatieven.

>  Volgens critici houdt de overheid dat duopolie juist in stand door apps slechts aan te bieden bij deze twee bedrijven. Terwijl er alternatieven zijn. Zo zijn er verschillende Google-vrije besturingssystemen gebaseerd op Android. Ook zijn er apps in de 'open source'-winkel F-Droid.

> De overheid zou alternatieven juist moeten stimuleren, zegt de bij F-Droid betrokken ontwikkelaar Hans-Christoph Steiner. 'Als overheden ook ons logo naast hun app zouden plaatsen, dan zou dat voor ons een geweldig verschil maken', zegt Steiner.

**Comment by Thomas-N** [comment]
Any news on this? This issue has been open for a long time now without any official comment from the developers.

I hope minvws is not waiting for parliament to demand this should be done, like it did with [Motie-Gerkens](https://www.eerstekamer.nl/motiedossier/35538_j_motie_gerkens_sp_c_s_over)*, as that would mean waiting until september at least.

*Which unfortunately did not have the intended result in the end, but this time around the technological hurdles involved seem to be a lot easier to overcome.

**Comment by obfusk** [comment]
> This issue has been open for a long time now without any official comment from the developers.

The lack of *any* response is not a good look.

**Comment by obfusk** [comment]
FWIW: I've heard (but not tested this myself) that -- now that Switzerland and the EU decided to recognise each other's covid certificates -- EU citizens can use the Swiss covid app -- which *is* available via F-Droid -- by importing/scanning a QR code generated via the website.

**Comment by pjvleeuwen** [comment]
> FWIW: I've heard (but not tested this myself) that -- now that Switzerland and the EU decided to recognise each other's covid certificates -- EU citizens can use the Swiss covid app -- which _is_ available via F-Droid -- by importing/scanning a QR code generated via the website.

Dank voor de tip. Ik denk dat je deze app bedoeld : https://github.com/admin-ch/CovidCertificate-App-Android

Let op, er is een 'wallet' (to store certificates) en verifier (to check them) app. Als ik op dit moment probeer de "COVID Certificate" app te installeren, dan krijg ik "COVID Certificate Check" en vice versa. Ook de share hyperlink voor de check laat zien dat het eigenlijk de wallet is. Dit is voor mij op dit moment de juiste link voor de wallet : https://f-droid.org/packages/ch.admin.bag.covidcertificate.wallet

Ik kan bevestigen dat deze wallet de EU / internationale QR code duidelijk kan opslaan (welke op te vragen is op coronacheck.nl )
Deze app werkt op mijn GrapheneOS telefoon (zonder Google services framework). De app begint met een paar pagina's informatie (next, next, next) welke internet nodig hebben. Daarna kan de internet permissie worden ingetrokken bij het scannen van de QR code en dit werkt goed tot je de app opnieuw probeert op te starten, dan is er weer internet verbinding nodig, anders crashed de app. Meh, ik gebruik de PDF en heb de app geinstalleerd voor het geval een instantie deze per se in de app wil zien.

**Comment by ijansch** [comment]
Thanks for the suggestion. Shipping the app was our top priority, so we first focused on the stores that would help the most users. We will evaluate this feature request and will update the ticket in the next couple of weeks.

**Comment by ijansch** [comment]
As for the Swiss app mentioned in the comments. The agreement pertains to the ability to scan and recognize each other’s certificates. A Dutch citizen can still use the coronacheck app to show their qr in Switzerland. You don’t need to install other country’s apps. Likewise. Swiss person coming to nl can be scanned with the Dutch scanners while showing the DCC in their Swiss app.

**Comment by obfusk** [comment]
> You don’t need to install other country’s apps.

True. But the Swiss one is available via F-Droid whereas the Dutch one is not. And both apps should work in multiple countries.

**Comment by obfusk** [comment]
> Thanks for the suggestion. Shipping the app was our top priority, so we first focused on the stores that would help the most users. We will evaluate this feature request and will update the ticket in the next couple of weeks.

Thank you.

Feel free to ping me if I can help with anything related to F-Droid (or reproducible builds) :)

**Comment by pjvleeuwen** [comment]
Thanks for the reply. Just making the APK available on the coronacheck website would be a good and easy start to break the dependency on accounts from specific companies (would be my guess). For your consideration.

**Comment by obfusk** [comment]
I've briefly looked at how easy it would be to get these apps into F-Droid.

Unfortunately, it seems they use the proprietary mlkit (which in turn depends on gms and firebase), which would violate our inclusion policy. I'm not sure whether this means they would not even work on devices without play services.

It would of course still be great to have the APKs here on github (or provided via a custom F-Droid repository).

**Comment by Thomas-N** [comment]
> I've briefly looked at how easy it would be to get these apps into F-Droid.
> 
> Unfortunately, it seems they use the proprietary mlkit (which in turn depends on gms and firebase), which would violate our inclusion policy. I'm not sure whether this means they would not even work on devices without play services.
> 
> It would of course still be great to have the APKs here on github (or provided via a custom F-Droid repository).

The correct solution would be to rewrite those parts of the software to no longer depend on mlkit. This app serves an important public need and is funded completely with public money. If at all possible, it should not have proprietary, closed source dependencies.

Other countries have already shown that it _is_ possible to write similar apps without these proprietary dependencies.

**Comment by obfusk** [comment]
I've created a patch that removes/replaces the proprietary parts (at least the obvious ones that were autodetected by `fdroid scanner`, someone still needs to do some manual checking).

I've proposed two (draft) merge requests for f-droid: [holder](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9580) & [verifier](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9581). I've only briefly tested the patched apps (help wanted!). The CI should produce an unsigned APK you can download.

Ideally these changes would be integrated here, or at least provided as an "fdroid" flavour.

**Comment by obfusk** [comment]
Unfortunately, it seems that all of the images used in these apps are 100% proprietary and cannot be redistributed by anyone (except for GitHub and Google) without explicit permission.

> It is therefore not permitted to reuse or transfer the image, unless the image explicitly states that it is permitted or you have received written permission of the Dutch Government. Such permission will not be withheld on unreasonable grounds.

I'd like to hereby request permission for f-droid to use them, but I won't be holding my breath.

In the mean time, the merge requests I've now closed do contain a patch that anyone should be able to use to build a FOSS (apart from the images) version of this app for themselves using the source code found here, which is far from user-friendly but at least something.

**Comment by obfusk** [comment]
Since GitHub is clearly allowed to redistribute the images, I've provided unsigned APKs of my FOSS version (built by GitHub Actions) [here](https://github.com/obfusk/nl-covid19-coronacheck-app-android/releases/tag/2.2.1-foss).

**Comment by obfusk** [comment]
> I'd like to hereby request permission for f-droid to use them.

cc @ijansch

**Comment by obfusk** [comment]
FSFE press release about the situation in Germany: https://fsfe.org/news/2021/news-20210830-01.en.html

**Comment by obfusk** [comment]
Since we still haven't gotten permission, I've removed the images and re-opened the merge requests:
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9580
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9581

**Comment by Enslinn** [comment]
Wat een opluchting dit topic van Derkades gevonden te hebben en te merken dat ik niet de enige Nederlander ben die een Google/ Apple vrije app wil! Ik vond het al heel vreemd dat de CoronaCheck app niet op F-Droid te vinden was. Temeer de Nederlandse politici zich weer op de borst aan het kloppen waren hoe goed ze het geregeld hadden: ziehier de open source app. Niet dus! 
Op: https://coronacheck.nl/nl/faq staat onder 'Ik heb gehoord dat de code open source is. Waar kan ik die vinden?' dit fabeltje nog eens.
De meeste politici hebben geen benul van IT en daar wordt steeds weer misbruik van gemaakt door IT'ers die onze privacy schenden want 'de Google producten zijn zo makkelijk te gebruiken'. Niet alleen schandalig maar ook een doorgaande bedreiging van onze westerse democratie. 

Nee wij zijn niet geïnteresseerd in het maken van een Google of Apple account. Wij willen alleen die CoronaCheck app.

Er is geen enkele democratiese reden te bedenken dat deze Big Tech criminelen het alleenrecht op distributie hebben.
Nu het bezit van deze app vereist gaat worden voor het toegang krijgen tot diverse (openbare) plekken word het des te dringender dat ie vrij komt voor F-Droid. 
Hij is ontwikkelt met ons belastinggeld en dient niet door 2 Big Tech bedrijven gekaapt te worden. Zelfs het Financieel Dagblad pleit daar voor.
Fantastiese bijdrage van obfusk!

@pjvleeuwen  Je dient natuurlijk een 'veilige' printer ter beschikking te hebben omdat de pdf ook je Internationaal bewijs bevat.  De printversie heeft ook dit nadeel: 'De papieren versie biedt daarom minder privacy. Door regelmatig een nieuwe QR-code te printen kun je dit voorkomen.' https://coronacheck.nl/nl/faq/5-2-wat-is-het-verschil-tussen-het-coronabewijs-op-papier-en-in-de-coronacheck-app/

So bloody please set this app free.

**Comment by obfusk** [comment]
The FOSS versions of the apps (sadly still without some of the images -- still working on getting official permission) are available on F-Droid: https://f-droid.org/packages/nl.rijksoverheid.ctr.holder/ & https://f-droid.org/packages/nl.rijksoverheid.ctr.verifier/

**Comment by diederikdehaas** [comment]
Thanks for all your work @obfusk :+1: 
After I realized I should give the app permission in AFWall+, it worked flawlessly :-)

**Comment by Enslinn** [comment]
Alles leuk en aardig maar die vervloekte DIGI D app zit ook niet in f-droid! Realiseer mij dat nu pas want gebruik DIGI D altijd op de pc. Daarnaast gebruik ik smartphone nooit online. 
Is daar nooit tegen geprotesteerd?
Kortom hoe krijgen wij een DIGI D app open source en in f-droid?

**Comment by TheLastProject** [comment]
The lack of DigiD app is indeed annoying, but it seems you don't really really need it as without the app you can use the webpage and use SMS-check.

As far as I know DigiD is completely proprietary, but the promising alternative [IRMA](https://f-droid.org/en/packages/org.irmacard.cardemu/) is on F-Droid. Now let's hope the government will finally support that too...

**Comment by Enslinn** [comment]
Als je een print wilt maken heb je digid nodig:
```
  Gegevens van je vaccinatie ophalen
Ben je al gevaccineerd? Log in met DigiD. De gegevens van je vaccinatie worden dan automatisch opgehaald.
```https://coronacheck.nl/nl/print/vaccinatie-ophalen

Lijkt mij dat je dat ook voor de coronacheck app nodig hebt.

**Comment by TheLastProject** [comment]
You have to login through DigiD. There are multiple ways to log into that. One of them is through the proprietary DigiD app, but you can also use SMS verification for example. So you don't need the DigiD app to use the CoronaCheck app.

**Comment by Enslinn** [comment]
hmmm... heb je dat getest?

**Comment by TheLastProject** [comment]
Well there was a SMS option. But sure, I tested it just now for you and yes, SMS confirming through the website works fine. You do not need the proprietary DigiD app.

**Comment by diademiemi** [comment]
I want to add to this that the app does work without Google Play Services. You can use the [Aurora Store](https://f-droid.org/en/packages/com.aurora.store/) to download the CoronaCheck app, and log in as normal. Logging in with DigiD opens the login page in the browser, and redirects back to the app after. If like me you don't have a Google account and don't want to install Google Play Services (or [MicroG](https://microg.org/)) this seems to work without issues, albeit needlessly complicated.

I still believe the CoronaCheck app should be on F-Droid regardless, since this is only a workaround, which still downloads it from Google. CoronaCheck was made with tax money, I do not believe it should be dependent on Google and Apple. This app should be available on as many platforms as possible, and I believe it should be completely open source and not utilize any proprietary components, which potentially put users privacy at risk.

**Comment by obfusk** [comment]
> Well there was a SMS option. But sure, I tested it just now for you and yes, SMS confirming through the website works fine. You do not need the proprietary DigiD app.

Correct. If you don't have the DigiD app installed, you can log in on your phone using a browser and authenticate using SMS.

DigiD plans to remove SMS authentication though, which would make it impossible to log in to DigiD without the app, which is proprietary and only available via Apple/Google/Microsoft; see https://community.freedom.nl/t/overheidsbeleid-authenticatie-digid-uitsluiting-door-toepassing-big-tech/2116

**Comment by obfusk** [comment]
> I still believe the CoronaCheck app should be on F-Droid.

@diademiemi the FOSS version of CoronaCheck *is* available on F-Droid: https://f-droid.org/packages/nl.rijksoverheid.ctr.holder/

**Comment by diademiemi** [comment]
> @diademiemi the FOSS version of CoronaCheck _is_ available on F-Droid: https://f-droid.org/packages/nl.rijksoverheid.ctr.holder/
It's available on F-Droid yes, but it's not official and the description says it doesn't include all images. It's great that it's on there, and it definitely works, but I believe this should've been handled by the government and made available on F-Droid in the first place, not requiring others to put in the work to maintain and distribute that version.

**Comment by obfusk** [comment]
> It's available on F-Droid yes, but it's not official and the description says it doesn't include all images. It's great that it's on there, and it definitely works, but I believe this should've been handled by the government and made available on F-Droid in the first place, not requiring others to put in the work to maintain and distribute that version.

No argument there. I'm still trying to make that happen. But at least we have something that works *now*.

**Comment by diederikdehaas** [comment]
> Correct. If you don't have the DigiD app installed, you can log in on your phone using a browser and authenticate using SMS.

Indeed. That's how I did it.
I thought that I previously could just login with username and password though. (But I'm fine with SMS as 2FA method)

> DigiD plans to remove SMS authentication though, which would make it impossible to log in to DigiD without the app, which is proprietary and only available via Apple/Google/Microsoft; see https://community.freedom.nl/t/overheidsbeleid-authenticatie-digid-uitsluiting-door-toepassing-big-tech/2116

Thanks for informing me; I didn't know that.

And GVD :angry:

**Comment by Enslinn** [comment]
Op een smartphone met Los 17.1 updated 19 sept. CCapp 2.3- 2137 van Fdroid. Het lukt niet!
FAQ: 'you retrieve them with the app by entering the code provided by the test location.'
Op de inentings plek van de GGD enkel een A4 met de prik datums gekregen. Er staat nergens een 'code provided by the test location'

'You can also retrieve a vaccination record in the app.'
'apply for or activate a DigiD' en hup naar de DigiD site!
Kan ook geen screenshot v/d app maken: not allowed. 

Allemaal bijzonder ergerlijk! Het is de taak van de overheid om een goed functionerende Goolag vrije app te maken!

p.s. waarom is de app en de info in het Engels? Hij is bedoeld voor Nederlanders! Kan iemand aub uitleggen wat er hier mis gaat?

**Comment by TheLastProject** [comment]
I don't think this is the right place to ask for support, we're discussing here about having a version of the app available on F-Droid.

Right now it is, due to @obfusk's amazing work, but there are still a few issues:
- [ ] @obfusk is now maintaining code that, frankly, @minvws should be maintaining.
- [ ] The F-Droid version lacks several images due to @minvws still not having replied to https://github.com/minvws/nl-covid19-coronacheck-app-android/issues/24#issuecomment-900821772.
- [ ] The "default" version has proprietary dependencies, which in my view is not reasonable for a government app especially given the Netherlands is supposed to be following an "Open Source or explain" policy, which clearly is not being followed here.

@BartNijland91 @SanderVeer @ktiniatros @aiden-shi Given you seem to be the maintainers of this project, how about a response?

**Comment by obfusk** [comment]
I agree that this is not the right place to ask for support, but since you've already asked and I happen to have some answers...

> Op de inentings plek van de GGD enkel een A4 met de prik datums gekregen. Er staat nergens een _'code provided by the test location'_

If you got vaccinated via the GGD (instead of e.g. via your "huisarts"), you need to use DigiD to generate a QR code in the app (or on the website if you want to print one instead). Which you can do with the F-Droid version as well. The app will open a browser window for you to log in (unless you have the DigiD app); I've seen some issues here on GitHub regarding which browser(s) it will detect and try to use, so it may not work flawlessly "out of the box".

> p.s. waarom is de app en de info in het Engels? Hij is bedoeld voor Nederlanders! Kan iemand aub uitleggen wat er hier mis gaat?

The description for the F-Droid version is only in English, because I only provided en-US metadata for F-Droid (so far). The app is both English and Dutch, but uses the default language for your phone. If I set my phone to Dutch, I get the app in Dutch.

**Comment by Enslinn** [comment]
Bedankt obfusk voor je soepele hulpvaardige houding. Vraag is waar je dan in godsnaam support kan/ mag vragen voor deze onafhankelijke CC app versie? Bij de NL overheid iig niet. Of is het de bedoeling dat dit een prive feestje voor IT techneuten blijft?

Ondanks al het werk van obfusk en alle uren die ik zelf, als 'gewone' Nederlander, hier al aan besteed heb zit ik nog steeds in de val van Goolag cs.

Daarom permitteer ik mij de vraag toch te stellen. Bedoelen jullie de sms-code die gestuurd is via https://login.digid.nl/sms_controleren in te voeren bij CCapp?

**Comment by obfusk** [comment]
> Bedankt obfusk voor je soepele hulpvaardige houding. Vraag is waar je dan in godsnaam support kan/ mag vragen voor deze onafhankelijke CC app versie? Bij de NL overheid iig niet. Of is het de bedoeling dat dit een prive feestje voor IT techneuten blijft?

For issues *specific to the F-Droid version* (like missing translations in the F-Droid client): https://github.com/obfusk/nl-covid19-coronacheck-app-android/issues

For anything not specific to the F-Droid version (which only replaces the QR scanner code and removes some images we can't legally distribute, but does not in any way modify the rest of the code -- like the digid integration; see the README for a full list of changes), you *should* be able to report issues here (or via the helpdesk); though maybe not in *this issue* :)

**Comment by obfusk** [comment]
> The F-Droid version lacks several images due to @minvws still not having replied to https://github.com/minvws/nl-covid19-coronacheck-app-android/issues/24#issuecomment-900821772

I also asked on the codefornl slack, which got me: ["We are currently investigating the possibility of offering an official build of the CoronaCheck app for the F-Droid store."](https://github.com/minvws/nl-covid19-coronacheck-app-coordination/blob/main/docs/Distribution.md) from @evavansloten; no updates on that for over a month :(

**Comment by diederikdehaas** [comment]
*I just have it set up the way I want, okay?!*

People who actually care and know about FLOSS will undoubtedly recognize it.
I highly doubt anyone at @minvws does, given the (mostly lack of) responses from them and just a cursory look at the profiles of the people who apparently have some official role in this.
Instead of harnessing the power of FLOSS, it seems (yet again) all about being able to put a checkmark next to 'Open Source'.

I've turned off notifications for this, so you'll have to mention me in case my input is wanted.

Thanks for all your great work @obfusk :+1: 
You obviously care and know about FLOSS :-)

**Comment by FintasticMan** [comment]
I've just noticed commit 5f6c0ed2e81281c5163a3a42d832273ef7696e37, which suggests official F-Droid support has been added. How does this affect availability on the main F-Droid repository?

**Comment by TheLastProject** [comment]
I just reinstalled my phone and found out the hard way that the F-Droid version no longer works. After logging into DigiD, it lists your certificates but when adding them it returns error A 270 000 401 99940.

I am extremely disappointed that after all this time and all the work from **unpaid volunteers** still nothing has happened from the @minvws side.

It has taken so long that we don't even use passes in the Netherlands anymore so the app is only useful for international travel.

This is frankly **disgraceful** and shows a huge lack of care about the both the **health of Dutch citizens** and **digital sovereignty.** It is **unbelievable** that a government that accepted an "Open Source by default" law hasn't lifted **a single finger** while the Open Source community has done all the hard work for them. It is disgusting this government has let the work of volunteers submitted to them **deteriorate** to the point they no longer work, **actively forcing citizens to be dependent on US tech giants** [as FD so clearly stated](https://fd.nl/economie-politiek/1383007/overheid-dwingt-burger-in-de-richting-van-google-en-apple).

**When will this app finally be made properly Open Source and submitted for inclusion to F-Droid? Will we have to wait until international certificates stop existing too and the app literally loses all purpose before a single finger will be lifted?**

**Comment by royarisse** [comment]
@TheLastProject It's good to see you're concerned about *digital sovereignty*. From that point of view,
I think this project shouldn't even exist.

For example, the client side of these apps are all open sourced, but what about the back-end? No information is known about any infrastructure of DigiD or Minvws, so we have absolutely no idea what is logged when ans where. Neither do we know what other countries policies on logging are. And the idea of showing a QR with quite some PII* to unknown — thus untrustworthy — other phones…

Finally, what's wrong about just trusting your citizens as a government/country? I think we're perfectly capable of staying at home when we're sick, we don't need no app for that. I mean, we, as citizens, are ought to trust the government as well, right?

Short story long, I'm probably stating the obvious, but I feel you shouldn't install this app.

*) One might argue the Domestic QR has almost no info at all, but all it's information, combined with for example when and where you're being scanned can aggregate into PII. For the European QR it's quite obvious it's PII.

**Comment by obfusk** [comment]
> I've just noticed commit [5f6c0ed](https://github.com/minvws/nl-covid19-coronacheck-app-android/commit/5f6c0ed2e81281c5163a3a42d832273ef7696e37), which suggests official F-Droid support has been added. How does this affect availability on the main F-Droid repository?

So... I've felt quite bad about having been unable to maintain my FOSS fork (which is why it is no longer in F-Droid) and today I find out that:

* there has been an official F-Droid flavour since November 2021;
* which cannot be used by F-Droid since the images in the app are still not redistributable;
* which no one ever told me or anyone at F-Droid about;
* which uses copy-pasted code of mine from the FOSS fork (great!);
* but without providing any attribution (which seems like a licence violation to me).

Anyone care to explain any of this?

**Comment by diederikdehaas** [comment]
@obfusk you shouldn't feel bad, it's @minvws that should feel bad about doing such a horrible job.
The (few) times I had to use your app, it worked flawlessly, so thanks :+1: 

> - which uses copy-pasted code of mine from the FOSS fork (great!);
> - but without providing any attribution (which seems like a licence violation to me).

IANAL but that does seem like a license violation to me (paragraph 5 of LICENSE.txt)

I saw your repo/fork has been archived, but I didn't see the reason why (I assume bc of what you said above).
Is there a new location of the repo? Anything I/others can do to help?

**Comment by obfusk** [comment]
>  Is there a new location of the repo? Anything I/others can do to help?

I can't work on my fork any more. So someone would have to create a new fork (to remove the images) and maintain it. Or get the images licensed in a way they can be distributed so the F-Droid flavour can be used as-is.

At least the code changes now seem to be available in the F-Droid flavour here. But the images -- which I mostly removed in my fork (though I did replace the app logo since that wasn't optional) -- are still not redistributable.

> It is therefore not permitted to reuse or transfer the image, unless the image explicitly states that it is permitted or you have received written permission of the Dutch Government. Such permission will not be withheld on unreasonable grounds.

Personally, I consider simply never responding to my request a form of withholding permission on unreasonable grounds. But I'm not a lawyer.

It' almost funny that this official repository now [claims to be my FOSS fork instead](https://github.com/minvws/nl-covid19-coronacheck-app-android/blob/main/holder/src/main/play/listings/en-US/full-description.txt) in its metadata:

> NB: this is the FOSS version for F-Droid, which removes non-free images and proprietary dependencies, and uses zxing to scan barcodes instead.

Because that was also simply copy-pasted it seems.

**Comment by royarisse** [comment]
@minvws doesn't really seem to understand what Open-Source stands for…
As stated in an earlier comment, working in an offline repository and pushing to GitHub once every new moon, resulting in so many changes no-one is able to keep up is not as it should be. And now they blatantly copy-paste someone else's code without mentioning the source, that's just rude.

I don't really understand why this project is still worked on anyways, can we just call this experiment done? :)

**Comment by ijansch** [comment]
Thanks for all your input on this subject. We are currently evaluating if and how we can roll out f-droid support.

We have done a previous attempt, but due to changes in priority that wasn't completed back then. In that effort we have indeed looked at the work that was already put into this by the community. If we continue on this and roll out an 'endorsed' fdroid version, and make use of the efforts you've already put in, then we will of course credit the respective contributors for their work. (if desired; some contributors in the past have chosen to not have their name publicly tied to the project given the discussion in society around the application of vaccination proofs).

I will soon update this Github issue with the status of this topic.

**Comment by ijansch** [comment]
Meanwhile the Minister has decided [^1] that continuing development or making the app available on other platforms is no longer relevant given the current state of the pandemic. I will close this ticket now. 

[^1]:https://www.rijksoverheid.nl/documenten/kamerstukken/2023/04/25/kamerbrief-over-diverse-onderwerpen-met-betrekking-tot-het-covid-19-virus

**Comment by confiks** [comment]
@obfusk Can you indicate which source files have been copied into the public repository without attribution? I've reviewed the diff [here](https://github.com/minvws/nl-covid19-coronacheck-app-android/compare/main...obfusk:nl-covid19-coronacheck-app-android:foss), and there's primarily the code in [`QRCodeScannerFragment.kt`](https://github.com/minvws/nl-covid19-coronacheck-app-android/compare/main...obfusk:nl-covid19-coronacheck-app-android:foss#diff-8dac62ed9205f30697b2ae5d4c4a8005a38a3445c2a922a0527ff80528a84b0cR327) which now has become [`QrCodeProcessor.kt`](https://github.com/minvws/nl-covid19-coronacheck-app-android-private/blob/main/qrscanner/src/main/java/nl/rijksoverheid/ctr/qrscanner/QrCodeProcessor.kt). It's a pretty generic piece of code, but we will add a copyright notice crediting you to that file.

Are there any other files? Except for the first line, the `play/listings/nl-NL/full-description.txt` contains text copyrighted text by the Rijksoverheid that was part of the Play Store listing.

**Comment by FintasticMan** [comment]
I think all the changes from 5f6c0ed2e81281c5163a3a42d832273ef7696e37 were from obfusk.

**Comment by ijansch** [comment]
As mentioned since the project is no longer relevant we won’t proceed with the fdroid work. It wasn’t our intention to not credit. We are happy to credit the work that was copied even if it wasn’t finished or used in the end. If you answer to @confiks’s question we’ll fix those credits.

**Comment by obfusk** [comment]
If this app -- which many people had no choice but to rely on -- was supposed to be open source, then IMO it failed.

You had F-Droid maintainers willing to help bring the app to F-Droid and thus make it available to people without being forced to use the Play Store and accept Google's terms of service; however:

### The code may use an open-source license, but the images are not redistributable

> Finally – The EUPL 1.2 and the CC0 do not apply to photos, videos, infographics or other forms of images. It is therefore not permitted to reuse or transfer the image, unless the image explicitly states that it is permitted or you have received written permission of the Dutch Government. Such permission will not be withheld on unreasonable grounds.

I asked for permission several times but never received a reply. So the app had to be forked -- and looked much worse without the images -- for no good reason.

One could argue that permission was thus "withheld on unreasonable grounds".

### The code depends on proprietary libraries

I provided a patch for using zxing instead of mlkit. It was not accepted. So the app had to be forked for no good reason.

### There was zero engagement with the broader open source community

Apparently, there was work done to create an F-Droid flavour. But no one ever told me or anyone else at F-Droid. Nor was this work ever completed. Clearly this was not considered a priority.

And on top of that my code was simply copied without attribution. Sure, it wasn't much code, and I don't particularly care about the attribution for it. Especially now that the app is abandoned there isn't really much point.

What I do care about is the absolute lack of engagement with the people trying to help and the message that sends.
